patches/@endo+ses-ava+0.2.27.patch

```diff
@@ -0,0 +1,250 @@
+   'before',
+@@ -56,32 +71,35 @@ const testerMethodsWhitelist = [
+   'failing',
+   'serial',
+   'only',
+-  'skip',
+ ];
+ 
+ /**
+- * @param {TesterFunc} testerFunc
+- * @param {Logger} [logger]
+- * @returns {TesterFunc} Not yet frozen!
++ * @template {Function} T
++ * @param {T} testerFunc
++ * @param {Logger} [logError]
++ * @returns {T} Not yet frozen!
+  */
+-const wrapTester = (testerFunc, logger = defaultLogger) => {
+-  /** @type {TesterFunc} */
+-  const testerWrapper = (title, implFunc, ...otherArgs) => {
+-    /** @type {ImplFunc} */
++const augmentLogging = (testerFunc, logError = console.error) => {
++  const augmented = (title, implFunc, ...otherArgs) => {
+     const testFuncWrapper = t => {
+       harden(t);
+-      return logErrorFirst(implFunc, [t, ...otherArgs], 'ava test', logger);
++      return logErrorFirst(implFunc, [t, ...otherArgs], 'ava test', logError);
+     };
+     if (implFunc && implFunc.title) {
+       testFuncWrapper.title = implFunc.title;
+     }
+     return testerFunc(title, testFuncWrapper, ...otherArgs);
+   };
+-  return testerWrapper;
++  // re-use other properties (e.g. `.always`)
++  // https://github.com/endojs/endo/issues/647#issuecomment-809010961
++  Object.assign(augmented, testerFunc);
++  // @ts-expect-error cast
++  return augmented;
+ };
+ 
++// TODO check whether this is still necessary in Ava 4
+ /**
+- * The ava `test` function takes a callback argument of the form
++ * The Ava 3 `test` function takes a callback argument of the form
+  * `t => {...}`. If the outcome of this function indicates an error, either
+  * by throwing or by eventually rejecting a returned promise, ava does its
+  * own peculiar console-like display of this error and its stacktrace.
+@@ -105,25 +123,17 @@ const wrapTester = (testerFunc, logger = defaultLogger) => {
+  * that eventually rejects, the error is first sent to the `console` before
+  * propagating into `rawTest`.
+  *
+- * @param {TesterInterface} avaTest
+- * @param {Logger} [logger]
+- * @returns {TesterInterface}
++ * @template {Function} T Ava `test`
++ * @param {T} avaTest
++ * @param {Logger} [logError]
++ * @returns {T}
+  */
+-const wrapTest = (avaTest, logger = defaultLogger) => {
+-  const testerWrapper = /** @type {TesterInterface} */ (wrapTester(
+-    avaTest,
+-    logger,
+-  ));
+-  for (const methodName of testerMethodsWhitelist) {
+-    if (methodName in avaTest) {
+-      /** @type {TesterFunc} */
+-      const testerMethod = (title, implFunc, ...otherArgs) =>
+-        avaTest[methodName](title, implFunc, ...otherArgs);
+-      testerWrapper[methodName] = wrapTester(testerMethod);
+-    }
++const wrapTest = (avaTest, logError = console.error) => {
++  const sesAvaTest = augmentLogging(avaTest, logError);
++  for (const methodName of overrideList) {
++    sesAvaTest[methodName] = augmentLogging(avaTest[methodName]);
+   }
+-  harden(testerWrapper);
+-  return testerWrapper;
++  harden(sesAvaTest);
++  return sesAvaTest;
```

Comment:
IIUC, you're still returning a new augmented `sesAvaTest` rather than mutating the original `avaTest`. Yes? I don't mind. But didn't your PR comment suggest mutating the original in place?

Reply:
Correct, in this PR not mutating the original. [Clarified](https://github.com/Agoric/agoric-sdk/pull/5774#issuecomment-1186539828)